bittide/tests/Tests/Haxioms.hs

```diff
@@ -198,3 +199,18 @@ prop_lessThanMax = property $ do
   a <- forAll (genNatural c)
   b <- forAll (genNatural c)
   assert (c <= max a b)
+
+-- | Test whether the following equation (supplied by 'useLowerLimit') holds:
+--
+--     1 + n <= u
+--
+-- Given:
+--
+--     1 <= m, n + m <= u
+--
+prop_useLowerLimit :: Property
+prop_useLowerLimit = property $ do
+  u <- forAll (genNatural 0)
+  n <- forAll (genNatural 0)
+  m <- forAll (genNatural 0)
+  assert (1 > m || n + m > u || n + 1 <= u)
```

Comment:
Sure, why not. I am just curious that you can always rewrite every constraint this way.

Reply:
You could! :). I'm rewriting it this way because previously you'd (effectively) throw away a lot of test cases. I.e., we don't care about test cases whose inputs don't match the constraints anyway.